### PR TITLE
*: unify the http client

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
there are many HTTP clients in PD. It's not easy to maintain, we want to supports TLS if the server is configured.

### What is changed and how it works?
- init HTTP client in `server`
- replace `middleware` client
- replace `cluster` client
- replace `server/api` client
### Release note 
no need to write

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -310,6 +313,13 @@ func (s *Server) startEtcd(ctx context.Context) error {
 		}
 	}
 	s.client = client
+	s.httpClient = &http.Client{
```
Comment (rleungx) [inline server/server.go:316]: The client doesn't have a timeout which is different from the original behavior

Comment (nolouch) [reply server/server.go:316]: the new timeout is set up by `NewRequestWithContext`. you can check it.

Comment (nolouch) [reply server/server.go:316]: like https://github.com/pingcap/pd/pull/2368/files#diff-02ca8f1595ce93b75591586acd227047R1651-R1659

Comment (rleungx) [reply server/server.go:316]: Does https://github.com/pingcap/pd/pull/2368/files#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bR1182 also need a context?

File: server/api/util.go [hunk 2]
```diff
@@ -20,14 +20,17 @@ import (
...
-	"github.com/pingcap/pd/v4/server/cluster"
 	"github.com/pkg/errors"
 )
 
 var (
 
-	// dialClient used to dial http request.
-	dialClient        = cluster.DialClient
+	// testDialClient used to dial http request. only used for test.
+	testDialClient = &http.Client{
```
Comment (HunDunDM) [inline server/api/util.go:29]: Shouldn't variables used only in tests appear here?

**Pull request merged**